pyhap/accessory_driver.py

```diff
@@ -157,6 +181,68 @@ def __init__(self, *, address=None, port=51234,
         network_tuple = (self.state.address, self.state.port)
         self.http_server = HAPServer(network_tuple, self)
 
+    def start(self):
+        """Start the event loop and call `start_pyhap`.
+
+        Pyhap will be stopped gracefully on a KeyBoardInterrupt.
+        """
+        try:
+            logger.info('Starting the event loop')
+            self.add_job(self.start_pyhap)
+            self.loop.run_forever()
+        except KeyboardInterrupt:
+            self.loop.call_soon_threadsafe(
+                self.loop.create_task, self.async_stop())
+            self.loop.run_forever()
```

Comment:
Why do we need a second `run_forever` (first one on line 192)?

Reply:
The first one is so that `pyhap` doesn't stop, if all `run` methods are done. You should still be able to call `set_value` from a sensor readout for example or change a setting from the Home App.

The second one to run the `async_stop` method and I don't have to pass
```py
self.loop.call_soon_threadsafe(self.loop.create_task, self.async_stop())
```
to `self.loop.run_until_complete`. In the end both have the same effect.

Reply:
Yes, the first call is clear, my issue was with the second one. Any chance something else submits to the loop so that the loop doesn't stop even after the async_stop finishes? In that case, isn't `run_until_complete` a better choice just to be on the safe side?

Reply:
`async_stop` calls `loop.stop()` so it will terminate. Regarding other tasks, since the `Accessory.stop` methods can be `async` as well, it is even intended that other might add other calls to the event loop. The part missing here, which I will add later is to wait for all calls to finish before calling `stop`.